modules/gapi/src/compiler/gcompiler.cpp

```diff
@@ -176,6 +176,103 @@ namespace
     }
 } // anonymous namespace
 
+namespace validation
+{
+    class meta_matcher_visitor : cv::util::static_visitor<bool, meta_matcher_visitor>
+    {
+        struct meta_print_visitor : cv::util::static_visitor<bool, meta_print_visitor> {
+            std::ostream& out;
+
+            meta_print_visitor(std::ostream &output) :
+            out (output) {}
+
+            template<class MetaT>
+            bool visit(std::size_t, const MetaT &) {
+                out << cv::detail::meta_to_string<MetaT>();
+                return true;
+            }
+
+            bool visit (std::size_t, const cv::util::monostate &) {
+                out << "variant::monostate";
+                return true;
+            }
+        };
+
+    protected:
+        template<class Descr>
+        bool check(const cv::GMetaArg &meta)
+        {
+            bool ret = cv::util::holds_alternative<Descr>(meta);
+            if (!ret)
+            {
+                out << "Expected: " << cv::detail::meta_to_string<Descr>() << ", got: ";
+                meta_print_visitor v{out};
```

Comment:
hmm I see, its sort of prints type not the value -- actually, visitor is not required here at all? It is just `out <<  cv::detail::meta_to_string<MetaT>()` (except the case for `monostate` but that's workarountable)

Reply:
>cv::detail::meta_to_string<MetaT>()

Do you mean the second visitor?  it's required cause `out`  is variant too and we need expand it to print-out actual type we have (which actually in controversy with expected type), just for troubleshooting info in exception (user just read and change it)